src/state/stake/useFarmsPortfolio.ts

```diff
@@ -0,0 +1,212 @@
+
+  complexTokenAmounts?.forEach(tokenReward => {
+    const tokenSymbol = tokenReward.token?.symbol ?? ''
+    return complexRewardsFarmAmounts.hasOwnProperty(tokenSymbol)
+      ? (complexRewardsFarmAmounts[tokenSymbol].amount = complexRewardsFarmAmounts[tokenSymbol].amount.add(
+          tokenReward.amount
+        ))
+      : (complexRewardsFarmAmounts[tokenSymbol] = {
+          token: tokenReward.token,
+          amount: tokenReward.amount,
+          address: tokenReward.tokenAddr
+        })
+  })
+
+  const complexRewardsFriendlyFarmAmounts = Object.entries(complexRewardsFarmAmounts).map(([name, value]) => ({
+    token: value.token,
+    tokenSymbol: name,
+    amount: value.amount.toFixed(6),
+    address: value.address
+  }))
+
+  //
+  //
+  // Tri rewards
+
+  const pendingTriDataV1 = useSingleContractMultipleData(
+    farmsReady ? contractv1 : null,
+    'pendingTri',
+    farmsReady ? v1Farms.map(farm => farm.v1args) : []
+  )
+
+  const pendingTriDataV2 = useSingleContractMultipleData(
+    farmsReady ? contractv2 : null,
+    'pendingTri',
+    farmsReady ? v2Farms.map(farm => farm.v1args) : []
+  )
+
+  const allPendingTri = [...pendingTriDataV1, ...pendingTriDataV2]
+
+  const pendingTriLoading = callResultIsLoading(allPendingTri)
+
+  const earnedTriRewardPools = allPendingTri.length
+    ? allPendingTri.map(farmData => JSBI.BigInt(farmData?.result?.[0] ?? 0))
+    : []
+
+  const allEarnedTriAmounts = earnedTriRewardPools.length
+    ? earnedTriRewardPools
+        .map(poolReward => new TokenAmount(TRI[ChainId.AURORA], JSBI.BigInt(poolReward)))
+        .reduce((a: TokenAmount, b) => a.add(b))
+    : new TokenAmount(dummyToken, JSBI.BigInt(BIG_INT_ZERO))
+
+  const totalTriAmount = allEarnedTriAmounts
+
+  const stakingInfoData = useFetchStakingInfoData()
+
+  const userInfoV1 = useSingleContractMultipleData(
+    farmsReady ? contractv1 : null,
+    'userInfo',
+    farmsReady ? v1Farms.map(farm => farm.v1args) : []
+  )
+  const userInfoV2 = useSingleContractMultipleData(
+    farmsReady ? contractv2 : null,
+    'userInfo',
+    farmsReady ? v2Farms.map(farm => farm.v1args) : []
+  )
+
+  const allUserInfo = [...userInfoV1, ...userInfoV2]
+
+  const tokens = activeFarms.map(farm => farm.tokens)
+  const pairsResult = usePairs(tokens)
+
+  // Loading
+  if (complexRewardsLoading || pendingTriLoading || pairsResult.some(pair => pair[0] === PairState.LOADING)) return null
+
+  // Error
+  if (
+    callResultError(userInfoV1) ||
+    callResultError(userInfoV2) ||
+    callResultError(pendingComplexRewards) ||
+    callResultError(pendingTriDataV1) ||
+    callResultError(pendingTriDataV2) ||
+    !stakingInfoData ||
+    pairsResult.some(pair => pair[1] === null)
+  ) {
+    console.error('Failed to load staking rewards info')
+    return null
+  }
+
+  const totalStaked = activeFarms.map((_, index) => {
+    const { totalStakedInUSD, totalStaked } = stakingInfoData?.[index] ?? {}
```

Comment:
This might make things easier

```suggestion
    if (stakingInfoData?.[index] == null) {
      return BIG_INT_ZERO
    }

    const { totalStakedInUSD, totalStaked } = stakingInfoData[index]
```